### PR TITLE
Allow SAP admins to give reason for rejection

### PR DESCRIPTION
**Pull request by polybuildr**
Resolves T149.

### CODE REVIEW

**Comment by polybuildr** [comment]
{Resolves T149}.

File: src/app/models/sap_model.php [hunk 1]
```diff
@@ -294,17 +294,29 @@ public function get_task_submissions($user_id, $mission_id) {
...
 
-    public function submit_review($submission_id, $approved) {
+    public function submit_review($submission_id, $approved, $reason) {
         if ($approved) {
             $status = 1;
         } else {
             $status = 2;
         }
+
+        if ($reason) {
```
Comment (nisargjhaveri) [inline src/app/models/sap_model.php]: Is this really necessary? Can't you set reason to null?

Comment (polybuildr) [inline src/app/models/sap_model.php]: Good idea.

Comment (polybuildr) [inline src/app/models/sap_model.php]: That was rather stupid... Fixed now.

**Comment by polybuildr** [comment]
Updated PR to not contain stupid code. :P

File: src/app/views/sap/mission.php [hunk 2]
```diff
@@ -64,6 +64,12 @@
                     <p><?= nl2br(htmlspecialchars($task['description'])) ?></p>
                     <?php if (isset($task['submission']) && $task['submission']['done'] == 2): ?>
                         <p class="text-error">Your submission for this task was rejected. :/</p>
+                        <?php if (isset($task['submission']['reason'])): ?>
+                            <p>
+                                <strong>Reason given</strong>:
```
Comment (nisargjhaveri) [inline src/app/views/sap/mission.php]: Don't want a `<br>` after this line? The reason is longtext.

Comment (polybuildr) [inline src/app/views/sap/mission.php]: Meh, it looked fine when I was testing it. But yeah okay, adding `<br>`.

**Comment by polybuildr** [comment]
Updated. Added `<br>`.

**Comment by nisargjhaveri** [comment]
Looks good.

**Pull request merged**